apps/condo/domains/acquiring/schema/RegisterMultiPaymentService.js

```diff
@@ -0,0 +1,231 @@
+                })
+                const supportedBillingIntegrations = get(acquiringIntegration, 'supportedBillingIntegrations', [])
+                    .map(integration => integration.id)
+                const uniqueBillingIntegrationsIds = new Set(billingContexts.map(context => context.integration))
+                const unsupportedBillings = Array.from(uniqueBillingIntegrationsIds)
+                    .filter(integration => !supportedBillingIntegrations.includes(integration))
+                if (unsupportedBillings.length) {
+                    throw new Error(`${REGISTER_MP_UNSUPPORTED_BILLING} (${unsupportedBillings.join(', ')})`)
+                }
+
+                const billingIntegrations = await find('BillingIntegration', {
+                    id_in: Array.from(uniqueBillingIntegrationsIds),
+                })
+
+                const currencies = new Set(billingIntegrations.map(integration => integration.currencyCode))
+                if (currencies.size > 1) {
+                    throw new Error(REGISTER_MP_MULTIPLE_CURRENCIES)
+                }
+                const currencyCode = get(billingIntegrations, ['0', 'currencyCode'])
+
+                // Stage 3 Generating payments
+                const payments = []
+                for (let i = 0; i < groupedReceipts.length; i++) {
+                    const receiptsGroup = groupedReceipts[i]
+                    const consumer = consumersByIds[receiptsGroup.consumerId]
+                    const acquiringContext = acquiringContextsByIds[consumer.acquiringIntegrationContext]
+                    const groupReceipts = receiptsGroup.receiptsIds
+                    for (let j = 0; j < groupReceipts.length; j++) {
+                        const receipt = receiptsByIds[groupReceipts[j]]
+                        const frozenReceipt = await freezeBillingReceipt(receipt)
+                        const account = await getById('BillingAccount', receipt.account)
+                        const payment = await Payment.create(context, {
+                            dv: 1,
+                            sender: { dv: 1, fingerprint },
+                            amount: receipt.toPay,
+                            currencyCode,
+                            accountNumber: account.number,
+                            period: receipt.period,
+                            receipt: { connect: { id: receipt.id } },
+                            frozenReceipt,
+                            context: { connect: { id: acquiringContext.id } },
+                            organization: { connect: { id: acquiringContext.organization } },
+                        })
+                        payments.push(payment)
+                    }
+                }
+
+                const paymentIds = payments.map(payment => ({ id: payment.id }))
+                const totalAmount = payments.reduce((acc, cur) => {
+                    return acc.plus(cur.amount)
```

Comment:
It would be good to make a little techtalk about `big.js` and it's usage in our product. I think Subscriptions domain might put it in a use